exporters/otlp/CMakeLists.txt

```diff
@@ -30,10 +30,10 @@ if(WITH_OTLP_GRPC)
                         PROPERTIES EXPORT_NAME otlp_grpc_client)
   target_link_libraries(
     opentelemetry_exporter_otlp_grpc_client
-    PUBLIC opentelemetry_sdk opentelemetry_ext opentelemetry_proto)
+    PUBLIC opentelemetry_common opentelemetry_sdk opentelemetry_ext opentelemetry_proto)
 
   target_link_libraries(opentelemetry_exporter_otlp_grpc_client
-                        PRIVATE gRPC::grpc++)
+                        PUBLIC gRPC::grpc++)
```

Comment:
Why change this to PUBLIC link here?This should be PRIVATE link or all targets depend opentelemetry_exporter_otlp_grpc_client will link `gRPC::grpc++`, which will cause the problem in #1603 .

Reply:
Sorry, struggling to reproduce #1603...
Before spending much time on it, let me guess: Is this about linking against static gRPC libs by a chance?

Reply:
> Sorry, struggling to reproduce #1603... Before spending much time on it, let me guess: Is this about linking against static gRPC libs by a chance?

This BUG can be repreduce when compiling gRPC as static library and compiling otel-cpp as dynamic library on Linux(With ELF ABI).
In this situation, the global variable will shared the same address between dynamic libraries but the initialization guards is private, which will construct these global variables more than onece.

Reply:
@ays7  - Do you have further comments on above discussion ?

Reply:
Yes, I will try to look into static gRPC linking soon -- marked PR as WIP for now